lib/Test/PostgreSQL.pm

```diff
@@ -150,8 +164,10 @@ has psql_args => (
 );
 
 method _build_psql_args() {
-    return '-U postgres -d test -h 127.0.0.1 -p ' . $self->port
-         . $self->extra_psql_args;
+    return '-U postgres -d test -h '.
+        ($self->use_socket?$self->socket_dir:'127.0.0.1').
+        ' -p ' . $self->port
```

Comment:
I don't think you pass a port parameter if using a local unix file socket?
Can you check? I suspect you need to put a `if $self->use_socket` at the top of the subroutine, and then have two methods of building the args - the local socket would take a file-path, and the TCP socket method would take hostname and port.

Reply:
Yes, you have to pass a port number; the -h option is the socket directory, not the socket name, and the socket name includes the port number.